### PR TITLE
Add custom ID field and AfterLoad hook

### PR DESCRIPTION
**Pull request by awrichar**

### CODE REVIEW

File: pkg/dbsql/crud.go [hunk 1]
```diff
@@ -728,6 +740,12 @@ func (c *CrudBase[T]) getManyScoped(ctx context.Context, tableFrom string, fi *f
 		if err != nil {
 			return nil, nil, err
 		}
+		if c.AfterLoad != nil {
+			err = c.AfterLoad(ctx, inst)
```
Comment (peterbroadhurst) [inline pkg/dbsql/crud.go:744]: Feels still a bit weird that these fields will _not_ get propagated to an object passed into `Insert` or `Update`.

Don't think we can do anything about it, but wonder if any health warnings are needed?

**Review by peterbroadhurst — APPROVED** [review]
Approved with one comment to consider @awrichar

File: pkg/dbsql/crud.go (hunk 1 above)
Comment (awrichar) [reply pkg/dbsql/crud.go:744]: This ends up being most useful for performing final validation and formatting on the whole object after it is loaded (for things that cannot be done in GetFieldPtr). I can add a note to this effect. 

It is less useful for purely generated fields, for the reason you noted. Not sure if I need to add additional comments to specifically discourage that?

Comment (awrichar) [reply pkg/dbsql/crud.go:744]: I guess my unit test is now misleading in this regard.

Comment (awrichar) [reply pkg/dbsql/crud.go:744]: Added a comment and tweaked the unit test

**Pull request merged**